addons/web/static/src/core/assets.js

```diff
@@ -65,118 +67,196 @@ export const loadCSS = memoize(function loadCSS(url) {
+            }
         }
     }
-    return doc;
-}
+    return assets;
+});
 
 /**
- * Renders a public asset template and loads the libraries defined inside of it.
- * Only loads js and css, template declarations will be ignored. Only loads
- * scripts and styles that are defined in script src and link href, ignores
- * inline scripts and styles.
+ * Loads the given js/css libraries and asset bundles. Note that no library or
+ * asset will be loaded if it was already done before.
+ *
+ * @param {Object} desc
+ * @param {Array<string|string[]>} [desc.assetLibs=[]]
+ *      The list of assets to load. Each list item may be a string (the xmlID
+ *      of the asset to load) or a list of strings. The first level is loaded
+ *      sequentially (so use this if the order matters) while the assets in
+ *      inner lists are loaded in parallel (use this for efficiency but only
+ *      if the order does not matter, should rarely be the case for assets).
+ * @param {string[]} [desc.cssLibs=[]]
+ *      The list of CSS files to load. They will all be loaded in parallel but
+ *      put in the DOM in the given order (only the order in the DOM is used
+ *      to determine priority of CSS rules, not loaded time).
+ * @param {Array<string|string[]>} [desc.jsLibs=[]]
+ *      The list of JS files to load. Each list item may be a string (the URL
+ *      of the file to load) or a list of strings. The first level is loaded
+ *      sequentially (so use this if the order matters) while the files in inner
+ *      lists are loaded in parallel (use this for efficiency but only
+ *      if the order does not matter).
+ * @param {string[]} [desc.cssContents=[]]
+ *      List of inline styles to add after loading the CSS files.
+ * @param {string[]} [desc.jsContents=[]]
+ *      List of inline scripts to add after loading the JS files.
  *
- * @deprecated
- * @param {string} xmlid The xmlid of the template that defines the public asset
- * @param {ORM} orm An ORM object capable of calling methods on models
- * @returns {Promise<void>} Resolved when the contents of the asset is loaded
+ * @returns {Promise}
  */
-export const loadPublicAsset = memoize(async function loadPublicAsset(xmlid, orm) {
-    const xml = await orm.call("ir.ui.view", "render_public_asset", [xmlid]);
-    const doc = new DOMParser().parseFromString(`<xml>${xml}</xml>`, "text/xml");
-    return Promise.all([
-        ...[...doc.querySelectorAll("link[href]")].map((el) => loadCSS(el.getAttribute("href"))),
-        ...[...doc.querySelectorAll("script[src]")].map((el) => loadJS(el.getAttribute("src"))),
-    ]);
+assets.loadBundle = memoize(async function loadBundle(desc) {
+    // Load css in parallel
+    const promiseCSS = Promise.all((desc.cssLibs || []).map(assets.loadCSS)).then(function () {
+        if (desc.cssContents && desc.cssContents.length) {
+            const style = document.createElement("style");
+            style.innerHTML = desc.cssContents.join('\n');
```

Comment:
Completely unnecessary, you can just set the *text* of the stylesheet.

Likewise, I assume, for the JS nodes.